ERCS/erc-7634.md

```diff
@@ -0,0 +1,223 @@
+
+*Expanding Use Cases*: These enhancements broaden the potential applications of NFTs by offering more control and flexibility to creators and owners. For instance, NFTs could be used to represent memberships or licenses with limited transferability, opening up new possibilities for digital ownership models.
+
+*Easy Integration*: To ensure broad adoption and ease of integration, this standard extends the existing [ERC-721](./eip-721.md) interface. By defining a separate interface (`IERC7634`) that includes the new functions, the standard allows existing [ERC-721](./eip-721.md) contracts to adopt the new features with minimal changes. This approach promotes backward compatibility and encourages the seamless incorporation of transfer limits into current NFT projects.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+- `setTransferLimit`: a function establishes the transfer limit for a tokenId.
+- `transferLimitOf`: a function retrieves the transfer limit for a tokenId.
+- `transferCountOf`: a function returns the current transfer count for a tokenId.
+
+Implementers of this standard **MUST** have all of the following functions:
+
+```solidity
+
+pragma solidity ^0.8.4;
+
+/// @title IERC7634 Interface for Limited Transferable NFT
+/// @dev Interface for ERC7634 Limited Transferable NFT extension for ERC721
+/// @author Saber Yu
+
+interface IERC7634 {
+
+    /**
+     * @dev Emitted when transfer count is set or updated
+     */
+    event TransferCount(uint256 indexed tokenId, address owner, uint256 counts);
+
+    /**
+     * @dev Returns the current transfer count for a tokenId
+     */
+    function transferCountOf(uint256 tokenId) external view returns (uint256);
+
+    /**
+     * @dev Sets the transfer limit for a tokenId. Can only be called by the token owner or an approved address.
+     * @param tokenId The ID of the token for which to set the limit
+     * @param limit The maximum number of transfers allowed for the token
+     */
+    function setTransferLimit(uint256 tokenId, uint256 limit) external;
+
+    /**
+     * @dev Returns the transfer limit for a tokenId
+     */
+    function transferLimitOf(uint256 tokenId) external view returns (uint256);
+}
+    
+```
+
+## Rationale
+
+### Does tracking the internal transfer count matter?
+
+Yes and no. It is **OPTIONAL** and quite depends on the actual requirements. The reference implementation given below is a **RECOMMENDED** one if you opt for tracking. The `_incrementTransferCount` function and related retrieval functions (`transferLimitOf` and `transferCountOf`) are designed to keep track of the number of transfers an NFT has undergone. This internal tracking mechanism is crucial for enforcing the minter's transfer limits, ensuring that no further transfers can occur once the limit is reached. 
```

Comment:
> **OPTIONAL**

The uppercase keywords create requirements on implementations. These keywords should only appear in the Specification section.

If you intend to create a requirement, move the text to the specification section. Instead, if you're simply discussing something and did not intend to create a requirement, use lowercase instead ("optional".)